docs/platform/howto/byoc/create-custom-cloud/create-google-custom-cloud.md

```diff
@@ -0,0 +1,491 @@
+       and are by default not accessible from outside. Traffic is routed through a proxy
+       for additional security utilizing a bastion host logically separated from the
+       Aiven services.
+   - `CLOUD_REGION_NAME` with the name of a Google region where to create your custom cloud,
+     for example `europe-north1`. See all available options in
+     [Google Cloud regions](/docs/platform/reference/list_of_clouds#google-cloud).
+   - `CIDR_BLOCK` with a CIDR block defining the IP address range of the VPC that Aiven
+     creates in your own cloud account, for example: `10.0.0.0/16`, `172.31.0.0/16`, or
+     `192.168.0.0/20`.
+   - `CUSTOM_CLOUD_DISPLAY_NAME` with the name of your custom cloud, which you can set
+     arbitrarily.
+
+    <details><summary>
+    Show sample output
+    </summary>
+
+    ```json
+    {
+        "custom_cloud_environment": {
+            "cloud_provider": "google",
+            "cloud_region": "europe-north1",
+            "contact_emails": [
+                {
+                    "email": "firstname.secondname@domain.com",
+                    "real_name": "Test User",
+                    "role": "Admin"
+                }
+            ],
+            "custom_cloud_environment_id": "018b6442-c602-42bc-b63d-438026133f60",
+            "deployment_model": "standard",
+            "display_name": "My BYOC Cloud on Google",
+            "errors": [],
+            "reserved_cidr": "10.0.0.0/16",
+            "state": "draft",
+            "tags": {},
+            "update_time": "2024-05-07T14:24:18Z"
+        }
+    }
+    ```
+
+    </details>
+
+1. Deploy the IaC template.
+
+   1. Download the template and the variable file:
+
+      - [avn byoc template terraform get-template](/docs/tools/cli/byoc#avn-byoc-template-terraform-get-template)
+
+            ```bash
+            avn byoc template terraform get-template        \
+              --organization-id "ORGANIZATION_ID"   \
+              --byoc-id "CUSTOM_CLOUD_ID" >| "tf_dir/tf_file.tf"
+            ```
+
+         Replace the following:
+
+         - `ORGANIZATION_ID` with the ID of your Aiven organization to
+           connect with your own cloud account to create the custom cloud,
+           for example `org123a456b789`. Get your `ORGANIZATION_ID`
+           [from the Aiven Console or CLI](/docs/platform/howto/byoc/create-custom-cloud/create-google-custom-cloud#prerequisites).
+         - `CUSTOM_CLOUD_ID` with the identifier of your custom cloud, which you can
+           extract from the output of the [avn byoc list](/docs/tools/cli/byoc#avn-byoc-list)
+           command, for example `018b6442-c602-42bc-b63d-438026133f60`.
+
+      - [avn byoc template terraform get-vars](/docs/tools/cli/byoc#avn-byoc-template-terraform-get-vars)
+
+            ```bash
+            avn byoc template terraform get-vars              \
+              --organization-id "ORGANIZATION_ID"     \
+              --byoc-id "CUSTOM_CLOUD_ID" >| "tf_dir/tf_file.vars"
```

Comment:
Our [internal documentation](https://coda.io/d/Product-Hub_di5xOUFIs2t/Project-Documentation_suttGL6t#Self-service-BYOC_tuMTGuNp/r2&view=center) and [test code](https://github.com/aiven/aiven-core/blob/6703e016a161792cd1cc369a6fdef25bb14eabc2/tests/systest_multicloud/custom_cloud_environments/test_custom_cloud_environment_google.py#L312) has been changed to recommend the `.tfvars` extension instead of `.vars`.

Hashicorp is [recommending](https://developer.hashicorp.com/terraform/language/values/variables#variable-definitions-tfvars-files) the `.tfvars` extension.
It enables proper [variable definition precedence](https://developer.hashicorp.com/terraform/language/values/variables#variable-definition-precedence).

Maybe we should do the same recommendation in our public doc.